From: Aurelien Jarno <aurel32@debian.org>
From: Cyril Brulebois <kibi@debian.org>
    
mesa fails to build on GNU/kFreeBSD, since some parts are not enabled.

Debian bug: http://bugs.debian.org/524690

--- a/configure.ac
+++ b/configure.ac
@@ -419,7 +419,7 @@
     case "$host_cpu" in
     i?86)
         case "$host_os" in
-        linux* | *freebsd* | dragonfly* | *netbsd* | openbsd*)
+        linux* | *freebsd* | dragonfly* | *netbsd* | openbsd* | kfreebsd*-gnu*)
             test "x$enable_64bit" = xyes && asm_arch=x86_64 || asm_arch=x86
             ;;
         gnu*)
@@ -429,7 +429,7 @@
         ;;
     x86_64|amd64)
         case "$host_os" in
-        linux* | *freebsd* | dragonfly* | *netbsd* | openbsd*)
+        linux* | *freebsd* | dragonfly* | *netbsd* | openbsd* | kfreebsd*-gnu*)
             test "x$enable_32bit" = xyes && asm_arch=x86 || asm_arch=x86_64
             ;;
         esac
@@ -962,7 +962,7 @@
             ;;
         esac
         ;;
-    *freebsd* | dragonfly* | *netbsd* | openbsd*)
+    *freebsd* | dragonfly* | *netbsd* | openbsd* | kfreebsd*-gnu*)
         DEFINES="$DEFINES -DHAVE_PTHREAD -DUSE_EXTERNAL_DXTN_LIB=1"
         DEFINES="$DEFINES -DHAVE_ALIAS"
 
